### DISCUSSION TITLE
Add CI tests against OpenFIPS201?

### DISCUSSION THREAD

**Discussion by no-usernames-left** [OpenSC/OpenSC#3165] [Ideas]
Hello,

According to the project's `README.md`, automated tests take place against a few applets, including `PIVapplet`.

Given how feature-complete [OpenFIPS201](https://github.com/makinako/OpenFIPS201) is, and the fact that its development has been commissioned and funded by the Australian Department of Defence (and will thus presumably be used by them), might it be a good idea to add that applet to the list which is automatically tested against?

**Comment by xhanulik** [comment]
I have already tried to run OpenFIPS201 applet as PIVApplet is run in automated tests, but I've encountered problem with loading the applet with jcardsim. The related issues are https://github.com/makinako/OpenFIPS201/issues/58 and https://github.com/licel/jcardsim/issues/205.

**Reply by frankmorgner** [reply]
Turns out, that OpenFIPS201 requires SCP03, which is not available in the free version of jCardSIM.

A quick and dirty solution would be to implement org.globalplatform.SecureChannel and org.globalplatform.GPSystem (which seem to be [the only classes needed for OpenFIPS](https://github.com/makinako/OpenFIPS201/blob/bd19f243b845aee174a8c19c14718f70b02b4d0d/src/com/makina/security/openfips201/OpenFIPS201.java#L34C8-L35)) with an empty implementation to just pretend that everything was secure in the CI environment. You could use [this project ](https://github.com/myasnikovsg/TestICCApplet/tree/master/src/org/globalplatform) as starting point, which provides the necessary definitions.

**Reply by frankmorgner** [reply]
See also https://github.com/makinako/OpenFIPS201/issues/68
